Update base url of direct links into RecordSearch

wragge opened the pull request:
The National Archives changed the location of the script that lets you build direct links into RecordSearch. This pull requests updates the urls to use the new redirect script.

dstillman commented on National Archives of Australia.js:
```diff
@@ -125,7 +125,7 @@ function scrape(doc, url) {
 			throw new Error("Unknown page type: " + m[1]);
 	}
-	
+	Zotero.debug(item);
```
Remove

wragge replied:
Done

dstillman commented on National Archives of Australia.js:
```diff
@@ -186,7 +186,7 @@ function scrapeItem(doc, url) {
 	
 	var barcode = encodeURIComponent(meta['item barcode']);
-	item.url = 'http://www.naa.gov.au/cgi-bin/Search?O=I&Number=' + barcode;
+	item.url = 'http://recordsearch.naa.gov.au/scripts/AutoSearch.asp?O=I&Number=' + barcode;
```
It looks like this site supports (and in fact redirects to) HTTPS, so these should all be changed to that.

wragge replied:
Done

wragge commented:
I've fixed those things, and checked against ESLint, but while I was doing that I found a few other things in need of fixing/improvement:

* the layout of PhotoSearch results & individual items had changed, so weren't being captured
* only digitised items were showing up when capturing item search results, and the titles weren't displaying
* series search results weren't being captured at all
* there's now a download link for PDF versions of digitised files that works! Added this in as well as some metadata enrichments for items and series

dstillman commented:
Thanks!
